### PR TITLE
Update to CUDA 1.0

### LINKED ISSUE #1228: CUDA.jl
Since

- [`CUDAnative.jl`](https://github.com/JuliaGPU/CUDAnative.jl)
- [`CUDAdrv.jl`](https://github.com/JuliaGPU/CUDAdrv.jl)
- [`CuArrays.jl`](https://github.com/JuliaGPU/CuArrays.jl)
- [`CUDAapi.jl`](https://github.com/JuliaGPU/CUDAapi.jl)

Are now all depreciated should we be switcing to [`CUDA.jl`](https://github.com/JuliaGPU/CUDA.jl)?

Does this require making KA use `CUDA.jl` as well?

### PR DESCRIPTION
**Pull request by jakebolewski**
# Description

The CUDA package / functionality has been significantly refactored for the 1.0 release.

KernelAbstractions needed to rename `CUDA` to `CUDADevice` for backend dispatch to resolve the symbol conflict, should be the only real major change as much of the GPU functionality is abstracted by KernelAbstractions.

Closes #1228 

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -42,15 +39,12 @@ Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
...
 
 [compat]
-Adapt = "1.0"
+Adapt = "1.0, 2.0"
 ArgParse = "1.1"
 CLIMAParameters = "= 0.1.4"
-CUDAapi = "3.1, 4.0"
-CUDAdrv = "6.0"
-CUDAnative = "2.10, 3.0"
+CUDA = "1.0"
```
Comment (vchuravy) [inline Project.toml:45]:
```suggestion
CUDA = "1.0"
KernelAbstractions = "0.3"
```

**Comment by simonbyrne** [comment]
You might need to run
```
julia --project=docs -e 'using Pkg; Pkg.resolve()'
```

**Comment by jakebolewski** [comment]
bors try

**Comment by simonbyrne** [comment]
The toml files need an update.

**Comment by jakebolewski** [comment]
Updated the packages, but still struggling with some kernel compilation issues on the GPU.

**Comment by simonbyrne** [comment]
bors try

**Comment by jakebolewski** [comment]
bors try

**Comment by jakebolewski** [comment]
Needs https://github.com/JuliaParallel/MPI.jl/pull/401

**Pull request merged**